### PR TITLE
Rewrite/repurpose/rename extension

### PR DESCRIPTION
**Pull request by Universal-Omega**

### CODE REVIEW

**Comment by BlankEclair** [comment]
Saw that the subdomain is trans, but it's not queer. Disappointed /j

File: includes/SubTranslate.php [hunk 1]
```diff
@@ -133,38 +123,26 @@ private static function callDeepL( $text, $tolang ) {
...
-		}
 
-		/* status here refers to the HTTP response code */
-		if ( $request->getStatus() !== 200 ) {
-			return "";
-		}
+		$requestFactory = MediaWikiServices::getInstance()->getHttpRequestFactory();
+		$request = $requestFactory->createMultiClient( [ 'proxy' => $wgHTTPProxy ] )
+			->run( [
+				'url' => 'https://trans.zillyhuhn.com/translate',
```
Comment (BlankEclair) [inline includes/SubTranslate.php]: Perhaps have a configuration option for specifying the LibreTranslate instance?

File: includes/SubTranslate.php [hunk 2]
```diff
@@ -133,38 +123,26 @@ private static function callDeepL( $text, $tolang ) {
...
-		if ( !$status->isGood() ) {
-			return "";
-		}
 
-		/* status here refers to the HTTP response code */
-		if ( $request->getStatus() !== 200 ) {
-			return "";
-		}
+		$requestFactory = MediaWikiServices::getInstance()->getHttpRequestFactory();
+		$request = $requestFactory->createMultiClient( [ 'proxy' => $wgHTTPProxy ] )
```
Comment (BlankEclair) [inline includes/SubTranslate.php]: Is manually setting the proxy necessary? https://doc.wikimedia.org/mediawiki-core/master/php/MWHttpRequest_8php_source.html#l00232

**Review by BlankEclair — COMMENTED** [review]
General comment: Mention using LibreTranslate in the README

File: includes/SubTranslate.php (hunk 2 above)
Comment (Universal-Omega) [reply includes/SubTranslate.php]: > Is manually setting the proxy necessary? https://doc.wikimedia.org/mediawiki-core/master/php/MWHttpRequest_8php_source.html#l00232

I thought so too, but it didn't work unless I did it like this.

File: includes/SubTranslate.php (hunk 1 above)
Comment (Universal-Omega) [reply includes/SubTranslate.php]: Done

**Pull request merged**